code/game/objects/items/robot/items/inflatable_tool.dm

```diff
@@ -0,0 +1,41 @@
+/obj/item/inflatable/cyborg
+	name = "inflatable wall"
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation."
+	var/power_use = 400
+	var/structure_type = /obj/structure/inflatable
+	var/delay = 1 SECONDS
+
+/obj/item/inflatable/cyborg/door
+	name = "inflatable door"
+	desc = "A folded membrane which rapidly expands into a simple door on activation."
+	icon_state = "folded_door"
+	power_use = 600
```

Comment:
Glass and metal are also generated from energy when you are in a recharger. It makes no difference whether it will be generated now or later, but the interface will be cluttered with extra lines + additional checks will need to be made not only for sheets of metal and glass, but also for “inflatable barriers”.
Using energy is an acceptable way.